### PR TITLE
fix(platform): implementation of expand and collapse all feature in table

### LINKED ISSUE #8901: fdp-table - implementation of expand and collapse all feature
#### Is this a bug, enhancement, or feature request?

feature request for platform treetable.

#### Briefly describe your proposal.

Need to implement expand and collapse all feature in table where we can expand and collapse all the rows of the table. 

#### Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)

current version

#### Is there anything else we should know?
Had discussion with cathy, Georg

### PR DESCRIPTION
**Pull request by skaquib123**
## Related Issue(s)

closes #8901 

## Description

Implemented expand and collapse all feature in table where we can expand and collapse all the rows of the table.

### Before:
image

##### During Implementation

1. Visual Testing:

-   [x] visual misalignments/updates
-   [x] check Light/Dark/HCB/HCW themes
-   [x] RTL/LTR - proper rendering and labeling
-   [x] responsiveness(resize)
-   [x] Content Density (Cozy/Compact/(Condensed))
-   [x] States - hover/disabled/focused/active/on click/selected/selected hover/press state
-   [x] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
-   [x] Mouse vs. Keyboard support
-   [x] Text Truncation

2. API and functional correctness

-   [x] check for console logs (warnings, errors)
-   [x] API boundary values
-   [x] different combinations of components - free style
-   [x] change the API values during testing

3. Documentation and Example validations

-   [x] missing API documentation or it is not understandable
-   [x] poor examples
-   [x] Stackblitz works for all examples

4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality

-   [x] the commit message(s) follows the guideline:
        https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
-   [x] tests for the changes that have been done
-   [x] all items on the PR Review Checklist are addressed :
        https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
-   [x] Run npm run build-pack-library and test in external application
-   [x] update `README.md`
-   [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

### CODE REVIEW

**Review by InnaAtanasova — APPROVED** [review]

File: libs/platform/src/lib/table/table.component.ts [hunk 1]
```diff
@@ -1443,8 +1465,13 @@ export class TableComponent<T = any> extends Table<T> implements AfterViewInit,
         const dropRowIndex = allRows.findIndex((row) => row === dropRow);
         const dropRowChildren = this._findRowChildren(dropRow);
 
-        const rowsBefore = allRows.slice(0, dropRowIndex + dropRowChildren.length + 1);
-        const rowsAfter = allRows.slice(dropRowIndex + dropRowChildren.length + 1);
+        if (dragRowIndex > dropRowIndex) {
```
Comment (platon-rov) [inline libs/platform/src/lib/table/table.component.ts:1468]: Why this block of code got changed? Is it related to expand/collapse feature?

File: libs/platform/src/lib/table/components/table-toolbar/table-toolbar.component.ts [hunk 2]
```diff
@@ -154,4 +157,14 @@ export class TableToolbarComponent implements TableToolbarWithTemplate, AfterVie
             this._showSaveButton = false;
         });
     }
+
+    /** @hidden */
+    expandAll(): void {
```
[the remark this answers is not in the document]
Comment (skaquib123) [reply libs/platform/src/lib/table/components/table-toolbar/table-toolbar.component.ts]: Changes done.

File: libs/platform/src/lib/table/components/table-toolbar/table-toolbar.component.ts [hunk 3]
```diff
@@ -52,6 +52,9 @@ export class TableToolbarComponent implements TableToolbarWithTemplate, AfterVie
     @Input()
     hideSearchInput = false;
 
+    /** Toggle to expand and collapse all feature */
```
Comment (g-cheishvili) [inline libs/platform/src/lib/table/components/table-toolbar/table-toolbar.component.ts:55]:
```suggestion
    /** Whether or not to show "expand all" and "collapse all" buttons */
```
I think for description, this fits better

**Review by g-cheishvili — APPROVED** [review]

**Review by mikerodonnell89 — COMMENTED** [review]
Works well, my only complaint is kind of a UX thing, if the table expand/collapse buttons are in the top half of the screen and i collapse then expand, the expansion goes upwards rather than down. I would think we would typically want to keep the expand/collapse buttons where they are when closing/expanding the table, if possible. I wonder if we need to track the scroll position of the buttons in the viewport and keep it in the same spot on expand/collapse. Not sure this is going to be doable, just something that felt a little strange when experimenting with this


https://user-images.githubusercontent.com/2471874/199274363-7a873c93-0496-4c80-9a62-6296ca93b4dd.mov

**Review by fkolar — APPROVED** [review]

**Pull request merged**